src/pymor/analyticalproblems/expressions.py

```diff
@@ -0,0 +1,447 @@
+class LE(BinaryOp):   numpy_symbol = '<='  # NOQA
+class GE(BinaryOp):   numpy_symbol = '>='  # NOQA
+class LT(BinaryOp):   numpy_symbol = '<'   # NOQA
+class GT(BinaryOp):   numpy_symbol = '>'   # NOQA
+
+
+class sin(UnaryFunctionCall):      numpy_symbol = 'sin'      # NOQA
+class cos(UnaryFunctionCall):      numpy_symbol = 'cos'      # NOQA
+class tan(UnaryFunctionCall):      numpy_symbol = 'tan'      # NOQA
+class arcsin(UnaryFunctionCall):   numpy_symbol = 'arcsin'   # NOQA
+class arccos(UnaryFunctionCall):   numpy_symbol = 'arccos'   # NOQA
+class arctan(UnaryFunctionCall):   numpy_symbol = 'arctan'   # NOQA
+class sinh(UnaryFunctionCall):     numpy_symbol = 'sinh'     # NOQA
+class cosh(UnaryFunctionCall):     numpy_symbol = 'cosh'     # NOQA
+class tanh(UnaryFunctionCall):     numpy_symbol = 'tanh'     # NOQA
+class arcsinh(UnaryFunctionCall):  numpy_symbol = 'arcsinh'  # NOQA
+class arccosh(UnaryFunctionCall):  numpy_symbol = 'arccosh'  # NOQA
+class arctanh(UnaryFunctionCall):  numpy_symbol = 'arctanh'  # NOQA
+class exp(UnaryFunctionCall):      numpy_symbol = 'exp'      # NOQA
+class exp2(UnaryFunctionCall):     numpy_symbol = 'exp2'     # NOQA
+class log(UnaryFunctionCall):      numpy_symbol = 'log'      # NOQA
+class log2(UnaryFunctionCall):     numpy_symbol = 'log2'     # NOQA
+class log10(UnaryFunctionCall):    numpy_symbol = 'log10'    # NOQA
+class sqrt(UnaryFunctionCall):     numpy_symbol = 'sqrt'     # NOQA
+class abs(UnaryFunctionCall):      numpy_symbol = 'abs'      # NOQA
+class sign(UnaryFunctionCall):     numpy_symbol = 'sign'     # NOQA
+
+
+class angle(UnaryFunctionCall):
+
+    numpy_symbol = 'angle'
+
+    def __init__(self, arg):
+        if arg.shape[-1] != 2:
+            raise ValueError
+        self.arg = arg
+        self.shape = arg.shape[:-1]
+
+
+class norm(UnaryReductionCall): numpy_symbol = 'norm'    # NOQA
+class min(UnaryReductionCall):  numpy_symbol = 'min'     # NOQA
+class max(UnaryReductionCall):  numpy_symbol = 'max'     # NOQA
+class sum(UnaryReductionCall):  numpy_symbol = 'sum'     # NOQA
+class prod(UnaryReductionCall): numpy_symbol = 'prod'    # NOQA
+
+
+class Pi(BaseConstant): numpy_symbol = 'pi'  # NOQA
+class E(BaseConstant):  numpy_symbol = 'e'   # NOQA
+
+
+pi = Pi()
+e  = E()
+
+
+_numpy_functions = {k: getattr(np, k) for k in {'sin', 'cos', 'tan', 'arcsin', 'arccos', 'arctan', 'arctan2',
+                                                'sinh', 'cosh', 'tanh', 'arcsinh', 'arccosh', 'arctanh',
+                                                'exp', 'exp2', 'log', 'log2', 'log10', 'sqrt', 'abs', 'sign',
+                                                'min', 'max', 'sum', 'prod',
+                                                'pi', 'e',
+                                                'array', 'broadcast_arrays', 'newaxis'}}
+
+_numpy_functions['norm']  = np.linalg.norm
+_numpy_functions['angle'] = lambda x: np.arctan2(x[..., 1], x[..., 0]) % (2*np.pi)
```

Comment:
Is this `% (2*np.pi)` also present in the numpy implementation of `angle`? After a short look into the [numpy code](https://github.com/numpy/numpy/blob/v1.21.0/numpy/lib/function_base.py#L1446-L1496) I don't think so. If I understand the [documentation of `arctan2`](https://numpy.org/doc/stable/reference/generated/numpy.arctan2.html) correctly, this should not change the result anyway.

Reply:
So I like my polar coordinate angles to go from 0 to 2*pi. This does not agree with the definition of `np.angle` but with the definition of `polar` we previously had for `ExpressionFunction`. So I'm not going to change the behavior in this PR. We can discuss this question in another issue however, if you wish. (The `% (2*np.pi)) is necessary, otherwise you get negative angles.)

Reply:
No, it's fine with me, no need to discuss this further. Maybe we should add a small note somewhere that the `angle` function behaves slightly different than the one in numpy.

Reply:
```suggestion
_numpy_functions['angle'] = lambda x: np.arctan2(x[..., 1], x[..., 0]) % (2*np.pi)  # np.angle uses different convention
```

Reply:
Done.